src/utils/data/vatglasses.ts

```diff
@@ -0,0 +1,698 @@
+    workerDataStore.vatglassesActiveAirspaces = vatglassesActiveAirspaces;
+    vatglassesUpdateInProgress = false;
+
+
+    console.timeEnd('updateVatglassesStateServer');
+    // console.log('activeVatglassesSectors', newActiveVatglassesPositions);
+    // console.log('vatglassesRunways', vatglassesRunways);
+    // console.log('activeSectors', activeVatglassesAirspaces);
+    // console.log('assignedVatglassesPositions', activeVatglassesPositions);
+}
+
+// This function is called at the first time combined data is needed. It fetches the combined data from the server and updates the local data. It is meant as an initial load of the combined data. Future updates and calculations are handled locally.
+let combineDataInitialized = false;
+async function initVatglassesCombined() {
+    console.log('initVatglassesCombined');
+    combineDataInitialized = true;
+    try {
+        const data: VatglassesActiveData = JSON.parse(await $fetch<string>(`/api/data/vatglasses-active`));
+        const vatglassesDataVersion = dataStore?.vatglasses?.value?.version;
+        if (vatglassesDataVersion === data.version) {
+            for (const countryGroupId in data.vatglassesActivePositions) {
+                for (const positionId in data.vatglassesActivePositions[countryGroupId]) {
+                    const serverPosition = data.vatglassesActivePositions[countryGroupId][positionId];
+                    const localPosition = dataStore.vatglassesActivePositions.value[countryGroupId]?.[positionId];
+
+                    if (localPosition) {
+                        if (serverPosition.airspaceKeys === localPosition.airspaceKeys) {
+                            localPosition.sectorsCombined = serverPosition.sectorsCombined;
+                            if (localPosition.lastUpdated) {
+                                localPosition.lastUpdated.value = new Date().toISOString();
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // Check existing runways with the runways from the server data and reset all changed icao codes
+        const serverRunways = data.vatglassesActiveRunways;
+        const localRunways = dataStore.vatglassesActiveRunways.value;
+
+        for (const icao in localRunways) {
+            if (localRunways[icao].active !== serverRunways?.[icao].active) {
+                await activeRunwayChanged(icao, false);
+            }
+        }
+        // Now call the update function to recalculate all sectors which were not updated by the server data or which had a different runway
+        updateVatglassesStateLocal();
+    }
+    catch {
+        console.error('Error fetching or processing vatglasses-active data');
+        // Optionally, you can handle the error further, such as displaying a user-friendly message
+    }
+}
+
+export async function initVatglasses(inputMode: string = 'local', serverDataStore: WorkerDataStore | null = null) {
```

Comment:
I had an issue when my browser froze every update even with vatglasses disabled. I suggest to not run any calculation in background and only do this when VG is enabled

Also it's weird that I had browser freeze while we have worker...

Reply:
The freeze is weird because the calculation only takes 20ms if nothing has changed. I have changed it that the calculations is only done when store.mapSettings.vatglasses?.active is active